frontend/src/metabase/questions/containers/ArchiveQuestionModal.jsx

```diff
@@ -14,6 +14,33 @@ const mapDispatchToProps = {
   archive: card => Questions.actions.setArchived(card, true),
 };
 
+const getLabels = question => {
+  const type = question.type() ?? "question";
+
+  if (type === "question") {
+    return {
+      title: t`Archive this question?`,
+      message: t`This question will be removed from any dashboards or pulses using it.`,
+    };
+  }
+
+  if (type === "model") {
+    return {
+      title: t`Archive this model?`,
+      message: t`This model will be removed from any dashboards or pulses using it.`,
+    };
+  }
+
+  if (type === "metric") {
+    return {
+      title: t`Archive this metric?`,
+      message: t`This metric will be removed from any dashboards or pulses using it.`,
+    };
+  }
```

Comment:
```suggestion
  const allowedTypes = ["question", "model", "metric"];
  
  if (allowedTypes.includes(type)) {
    return {
      title: t`Archive this ${type}?`,
      message: t`This ${type} will be removed from any dashboards or pulses using it.`,
    };
  }
```

WDYT about something like this?

Reply:
I intentionally avoided it. Dynamic strings come with some degree of magic.
With static strings it's easier to add translations and find translation keys using simple text search.